### PR TITLE
Refactor CSS; address some design issues

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Implement https://github.com/enso-org/cloud-v2/issues/924
  - Refactor all numbers out to CSS variables
- Implement some issues raised in the design meeting
  - The columns selector now only contains *hidden* columns, rather than all of them.
  - Unified opacity for active (100%), selectable and hovered (75%), selectable (50%) and disabled (30%)
    - Easily configurable if we want to change it in the future, so the specific values don't matter too much for now.
  - Always show asset right panel if it is enabled - display placeholder text if <1 or >1 asset is selected
  - Hide docs icon that was in the top right assets menubar (next to the gear icon for asset settings) (as backend functionality has yet to be implemented)
  - Clicking a user in the "Shared with" column now adds them to the search as `owner:<username>`
  - Add a gap between adjacent rows. This makes each row more visually distinct when many rows are selected
  - Center the left column (the first column) of the context menu below the mouse, rather than centering the entire context menu.
- Fix regressions caused by CSS refactor
  - Make keyboard selection indicator for asset rows rounded again
- Other misc. fixes and improvements
  - Slightly modified styling of chat reaction bar
  - Hide the row containing the "New Project" button in the cloud drive, when not in the "Home" drive tab
  - Animate rotation of column sort arrow when clicking on a column to change the sort order
    - Consistent duration of arrow rotation animation for folder arrows, column sort arrows, chat thread list arrows
    - Consistent icon for sort arrow for folders and the chat thread list
  - Minor adjustment of styles for optional properties in the Data Link input

Not included in this PR:
- Custom (HTML) scrollbars for consistency across all browsers and all OSes (except perhaps touchscreens)
  - Potentially time-consuming to look for a library (and not quite trivial to implement ourselves)
- Columns sliding left as they expand and right as they collapse
  - Also non-trivial, especially when taking into account horizontal scrolling.
- Fixing styles to closer resemble Figma design
  - As (kinda) mentioned in the meeting - ideally it should be pixel perfect, *but* value consistency with other spacings, opacities etc. over being 100% pixel-perfect
  - However, it has *partly* been done - mostly for the home page. It's entirely possible that changes made afterwards broke the spacing again though.

### Important Notes
None

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] ~~Unit tests have been written where possible.~~
  - [x] ~~If GUI codebase was changed, the GUI was tested when built using `./run ide build`.~~

### CODE REVIEW

File: app/ide-desktop/eslint.config.js [hunk 1]
```diff
@@ -231,7 +231,24 @@ const RESTRICTED_SYNTAXES = [
     {
         selector:
             'JSXElement[closingElement!=null]:not(:has(.children:matches(JSXText[raw=/\\S/], :not(JSXText))))',
-        message: 'Use self-closing tags (`<tag />`) for tags without children.',
+        message: 'Use self-closing tags (`<tag />`) for tags without children',
+    },
+    {
+        // TODO [sb]: `z-3` should be eliminated, but is currently still required.
+        // TODO: this does not work for template strings and nested strings
```
Comment (PabloBuchu) [inline app/ide-desktop/eslint.config.js]: is this gonna be fixed in this PR? if not lets make it as FIXME with gh issue

Comment (somebody1234) [reply app/ide-desktop/eslint.config.js]: cannot be fixed until GUI1 is removed, as `z-3` *is* the workaround for GUI1 - the reason being that GUI1 uses `z-1` through `z-4` for their renderinglayers. will add gh issue.

Comment (somebody1234) [reply app/ide-desktop/eslint.config.js]: note: the `TODO` on the second line is no longer accurate, so i've removed it

**Comment by PabloBuchu** [comment]
1. Isn't there any way to make this transition more smooth for the top right buttons groups? This blinking is annyoing.

https://github.com/enso-org/enso/assets/2855109/c7b78aed-5ec7-4907-bf2a-712b570bb4dc



2. When I have an opened running project and hover my pointer over `chat`, `share`, `setting` can we ged rid of this grey circle? 
3. When I open the settings menu.  Can we move the avatar to the right and align with the picture when the menu is closed? I would like the impression of menu toggling from the user picture
4. Not sure whether this PR is suitable: when I am using arrows to navigate I should be able to jump from files to recent/home/trash and switch to selected with enter (maybe even to the top buttons group as well?). Lemme know if you need separate issue for that

**Comment by somebody1234** [comment]
1. will do soon :tm:
2. probably not. the dashboard doesn't use fully transparent backgrounds, and this will be a non-issue in GUI2 since GUI2 doesn't use a custom cursor.
3. done, will push soon.
4. upon doing some thinking, it would probably be better to do that in another PR. two main changes need to happen:
  - the active tab (a `<button disabled />`, i.e. can't be clicked at all) needs to be selectable via keyboard
  - we also need a (preferably unified) way to navigate between all the areas. i assume we need something custom for *arrow keys* to work
    - however we should also select tab-based navigation (for screenreaders, for example). but this is a lower priority imo

**Comment by somebody1234** [comment]
okay... summary:
1. done
2. wontfix (probably)
3. done
4. should create new issue for this

**Comment by PabloBuchu** [comment]
you are reporting the 3 as done but I am still having the old look

Screenshot 2024-03-11 at 15 27 35

other things seems ok

**Comment by somebody1234** [comment]
that's intentional (see for example github) - i assume the icon being on the left is a conscious design choice.
i suppose one thing to note is that if the icon is on the right, we'd want the text to be right-aligned too. (and the menu entries will need icons on the right too!)

i'm assuming that the transition of the icon when the menu is opening would be a clear enough indicator would have the same effect as requested:
> I would like the impression of menu toggling from the user picture

although it's totally possible that i made the animation too fast, and so a (say) 50% slowdown (200ms -> 300ms) in the CSS transition might make it much nicer visually.

**Review by PabloBuchu — APPROVED** [review]

**Review by radeusgd — APPROVED** [review]
Approving the changes to the schema

**Pull request merged**